File: Array/linchuanxin/Algorithm/day60_0730_合并两个有序链表/day60_leetcode21.java
package day60_0730;




public class day60_leetcode21 {
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }


    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode dump = new ListNode(-1);
        ListNode dumphelper = dump;
        while(l1!=null && l2!=null){
            if(l1.val <= l2.val){
                dumphelper.next = l1;
                l1 = l1.next;
                dumphelper = dumphelper.next;
            }else{
                dumphelper.next = l2;
                l2 = l2.next;
                dumphelper = dumphelper.next;
            }
        }
        if(l1 == null)
            dumphelper.next = l2;
        if(l2 == null)
            dumphelper.next = l1;
        return dump.next;

    }

}

File: Array/linchuanxin/Algorithm/day60_0730_合并两个有序链表/day60_leetcode21.kt
package day60_0730

class ListNode(var `val`: Int) {
    var next: ListNode? = null
}


fun mergeTwoLists(l1: ListNode?, l2: ListNode?): ListNode? {
    var dump:ListNode? = ListNode(-1)
    var dumpHelper:ListNode? = dump
    var index_l:ListNode? = l1
    var index_r:ListNode? = l2
    while (index_l!=null && index_r!=null){
        if (index_l.`val` <= index_r.`val`){
            dumpHelper?.next = index_l
            index_l = index_l.next
            dumpHelper = dumpHelper?.next
        }else{
            dumpHelper?.next = index_r
            index_r = index_r.next
            dumpHelper = dumpHelper?.next
        }
    }
    if(index_l == null)
        dumpHelper?.next = index_r
    if(index_r == null)
        dumpHelper?.next = index_l
    return dump?.next
}
